ts/components/portfolio/PayLayout.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  Button,
+  Container,
+  Content,
+  H1,
+  Icon,
+  Left,
+  Right,
+  Text
+} from "native-base";
+import * as React from "react";
+import { Image, ScrollView } from "react-native";
+import { Col, Grid, Row } from "react-native-easy-grid";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import I18n from "../../i18n";
+import ROUTES from "../../navigation/routes";
+import { PortfolioStyles } from "../styles";
+
+export enum ImageType {
+  BANK_IMAGE
+};
+
+type Props = {
+  title: string,
+  subtitleLeft?: string,
+  subtitleRight?: string,
+  subtitle?: string,
+  touchableContent?: React.ReactElement<any>,
+  children?: React.ReactElement<any>,
+  rightImage?: ImageType,
+  navigation: NavigationScreenProp<NavigationState>
+  showPayNoticeButton?: boolean
+}
+
+// PayLayoutComponents
+enum PLC {
+  PAY_NOTICE_BUTTON, // pay notice button, anchored to the bottom of page
+  MAIN_SCREEN, // main screen, divided in two parts
+  TOP_PART, // top part of the main screen (title, some actions)
+  BOTTOM_PART, // bottom part of the main screen (contents e.g. transactions list)
+  TITLE_ROW, // title in the top part 
+  SUBTITLE_ROW, // subtitle in top part
+  TOUCHABLE_ROW, // touchable content in top part
+}
+
+
+/**
+ * Pay layout component
+ */
+export class PayLayout extends React.Component<Props, never> {
+
+  public render(): React.ReactNode {
+    return (
+      <Container>
+        <Grid>
+          <Row size={this.rowSize(PLC.MAIN_SCREEN)}>
```

Comment:
I don't understand the purpose of having the row sizes encapsulated inside `rowSize`, it makes understanding the structure harder (levels of indirection), besides `rowSize(PLC.MAIN_SCREEN)` is only used once